src/fhirtypes/ElementDefinition.ts

```diff
@@ -1642,8 +1643,16 @@ export class ElementDefinition {
    * @throws {InvalidUriError} when the system being assigned is not a valid uri
    */
   private assignFshCode(code: FshCode, exactly = false, fisher?: Fishable): void {
-    if (code.system && !isUri(code.system.split('|')[0])) {
-      throw new InvalidUriError(code.system);
+    if (code.system) {
+      const csURI = code.system.split('|')[0];
+      if (fisher) {
+        const vsURI = fisher.fishForMetadata(code.system, Type.ValueSet)?.url.split('|')[0] ?? '';
```

Comment:
Here we could probably just do:
```javascript
        const vsURI = fisher?.fishForMetadata(code.system, Type.ValueSet)?.url.split('|')[0] ?? '';
```
and avoid having the check for `fisher`.

Reply:
Addressed in `e8b6e3f`.